Mathlib/CategoryTheory/Iso.lean

```diff
@@ -140,6 +140,8 @@ def refl (X : C) : X ≅ X where
 
 instance : Inhabited (X ≅ X) := ⟨Iso.refl X⟩
 
+theorem nonempty_iso_refl (X : C) : Nonempty (X ≅ X) := ⟨default⟩ 
```

Comment:
Is this needed for something?

Reply:
No, things do seem to build without this lemma, shall I go ahead and remove it?

Reply:
Originally this https://github.com/leanprover-community/mathlib4/blob/c49e83649a214bc334ae6aa5bef9dd7518464b0c/Mathlib/CategoryTheory/Equivalence.lean#L798 was `by simp`. Using `simp?`, this was missing some inhabited to nonempty result. I thought I put this in there to provide `default` for a fix by hand. But if it builds fine, perhaps I am mistaken. Regardless, if it builds, it is fine to me.

Reply:
```
⟨e.symm Y, by simpa using ⟨default⟩⟩
```
seems closer to the mathlib3 version, and it should work just with the inhabited instance.

Reply:
Pushed this.